Scripts/MeshChainRenderer.cs

```diff
@@ -0,0 +1,335 @@
+    /// Set the color at the end of the line.
+    /// </summary>
+    public Color colorEnd
+    {
+        get { return m_Color.Evaluate(1); }
+        set
+        {
+            var colorKeys = m_Color.colorKeys;
+            var alphaKeys = m_Color.alphaKeys;
+            var lastColorIndex = colorKeys.Length - 1;
+            var lastAlphaIndex = alphaKeys.Length - 1;
+            var flatColor = value;
+            flatColor.a = 1.0f;
+            colorKeys[lastColorIndex].color = flatColor;
+            alphaKeys[lastAlphaIndex].alpha = value.a;
+            m_Color.colorKeys = colorKeys;
+            m_Color.alphaKeys = alphaKeys;
+            UpdateColors();
+        }
+    }
+
+    void OnValidate()
+    {
+        SetupMeshBackend();
+        if (NeedsReinitialize())
+        {
+            #if UNITY_EDITOR
+                UnityEditor.EditorApplication.delayCall += EditorCheckForUpdate;
+            #endif
+        }
+        else
+        {
+            EditorCheckForUpdate();
+        }
+    }
+
+    /// <summary>
+    /// Cleans up the visible interface of the meshrenderer by hiding unneeded components
+    /// Also makes sure the animation curves are set up properly to defualts
+    /// </summary>
+    void SetupMeshBackend()
+    {
+        m_MeshRenderer = GetComponent<MeshRenderer>();
+        m_MeshRenderer.hideFlags = HideFlags.HideInInspector;
+        var meshFilter = GetComponent<MeshFilter>();
+        meshFilter.hideFlags = HideFlags.HideInInspector;
+
+        if (m_Materials == null)
+        {
+            m_Materials = m_MeshRenderer.sharedMaterials;
+        }
+        m_MeshRenderer.sharedMaterials = m_Materials;
+
+        if (m_WidthCurve == null || m_WidthCurve.keys == null || m_WidthCurve.keys.Length == 0)
+        {
+            m_WidthCurve = new AnimationCurve(new Keyframe(0, 1.0f));
+        }
+        m_Color = m_Color ?? new Gradient { alphaKeys = new[] { k_DefaultStartAlpha, k_DefaultEndAlpha },
+                                             colorKeys = new[] { k_DefaultStartColor, k_DefaultEndColor },
+                                             mode = GradientMode.Blend };
+    }
+
+    /// <summary>
+    /// Makes the sure mesh renderer reference is initialized before any functions try to access it
+    /// </summary>
+    protected virtual void Awake()
+    {
+        SetupMeshBackend();
+        Initialize(true);
+    }
+
+    /// <summary>
+    /// Makes the sure mesh renderer reference is initialized on reset of the component
+    /// </summary>
+    void Reset()
+    {
+        SetupMeshBackend();
+        Initialize(true);
+    }
+
+    /// <summary>
+    /// Ensures the lines have all their data precached upon loading
+    /// </summary>
+    void Start()
+    {
+        Initialize();
+    }
+
+#if UNITY_EDITOR
+    /// <summary>
+    /// Ensures the hidden mesh/meshfilters are destroyed if users are messing with the components in edit mode
+    /// </summary>
+    void OnDestroy()
```

Comment:
I was seeing this error before but I believe I fixed it - can you update and see if it is still a problem?